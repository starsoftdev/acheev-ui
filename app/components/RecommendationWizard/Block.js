// @flow
import * as React from 'react';
import cx from 'classnames';

type Props = {
  children: React.Node,
  label: string,
  className?: string,
  selected?: boolean,
};

const Block = ({ children, label, className, selected, ...rest }: Props) => (
  <div
    className={cx('recommendationWizard__block', className, {
      'recommendationWizard__block--selected': selected,
    })}
    {...rest}
  >
    {children}
    <p className="recommendationWizard__label">{label}</p>
  </div>
);

export default Block;
