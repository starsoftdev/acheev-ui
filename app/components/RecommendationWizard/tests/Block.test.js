import React from 'react';
import { create } from 'react-test-renderer';
import Block from 'components/RecommendationWizard/Block';

describe('<Block />', () => {
  it('should match snapshot', () => {
    const component = create(<Block label="label">This go before label</Block>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot when selected', () => {
    const component = create(
      <Block label="label" selected>
        This go before label
      </Block>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
