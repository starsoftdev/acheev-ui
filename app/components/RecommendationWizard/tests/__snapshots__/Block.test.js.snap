// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Block /> should match snapshot 1`] = `
<div
  className="recommendationWizard__block"
>
  This go before label
  <p
    className="recommendationWizard__label"
  >
    label
  </p>
</div>
`;

exports[`<Block /> should match snapshot when selected 1`] = `
<div
  className="recommendationWizard__block recommendationWizard__block--selected"
>
  This go before label
  <p
    className="recommendationWizard__label"
  >
    label
  </p>
</div>
`;
